security/security-core/src/main/java/io/camunda/security/auth/OidcPrincipalLoader.java

```diff
@@ -58,5 +60,13 @@ private static String tryReadJsonPath(final Map<String, Object> claims, final Js
     }
   }
 
+  private String sanitizeClaimPath(final String claim) {
+    // If the claim starts with a dollar sign, it is already a JSONPath expression.
+    // Otherwise, we wrap it with the dollar sign to denote a JSONPath.
+    // We also ensure that the claim is wrapped in single quotes to handle cases where the claim
+    // name contains special characters.
+    return claim.startsWith("$") ? claim : "$['" + claim + "']";
```

Comment:
:thought_balloon: This will break for special characters such as `'`. If we really want to support this pattern long-term, the most reliable way would be to bypass JsonPath altogether.

Reply:
Yeah, you'd have to essentially turn `claim` into valid JSON, yeah? :thinking: Anyway, the chances of someone using `'` are much slimmer, I think (though of course, non-zero). The other option is escaping single quotes, e.g. `claim.replaceAll("'", "\\'")`. I'm sure there are still way to break things, but it probably works for 99% of cases - and if we have a good error message/reporting, it should be possible to tell users how to fix it on their side.